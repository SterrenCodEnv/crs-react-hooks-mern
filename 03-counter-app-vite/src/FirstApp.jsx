import PropTypes from 'prop-types';

export const FirstApp = ({ title, subTitle, author }) => {
  return (
    <>
      <h1 data-testid="test-title"> {title} </h1>
      <p>{subTitle}</p>
      <p>{author}</p>
    </>
  );
};

FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subTitle: PropTypes.string,
};

FirstApp.defaultProps = {
  title: 'No Title',
  subTitle: 'No SubTitle',
  author: 'Admin',
};
