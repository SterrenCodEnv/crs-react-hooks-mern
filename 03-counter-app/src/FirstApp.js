import React from 'react';
import PropTypes from 'prop-types';

const FirstApp = ({portMessage, portNumber}) => {

    const title = 'Data:'
    const data = {
        firstName: 'Juan',
        lasName: 'Sterren',
        email: 'sterrenjuanignacio@gmail.com',
        age: 30,
        job: 'Software Developer'
    }

    return (
        <>
            <h2>{title}</h2>
            <pre>{JSON.stringify(data, null, 4)}</pre>
            <p>{portMessage} {portNumber}</p>
        </>
    );
}

FirstApp.propTypes = {
    portMessage: PropTypes.string.isRequired
}

FirstApp.defaultProps = {
    portNumber: 3000
}

export default FirstApp;