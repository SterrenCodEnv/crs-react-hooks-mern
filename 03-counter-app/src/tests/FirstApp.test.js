import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import FirstApp from '../FirstApp';


describe('Test in FirstApp.js', () => {

    test('Show data JSON', () => {
        const wrapper = shallow(<FirstApp portMessage='Listening on port' portNumber='3000'/>)
        expect(wrapper).toMatchSnapshot();
    });

    test("portNumber return 'Listening on port 3000'", () => {
        const portMessage = 'Listening on port';
        const portNumber = '3000'
        const wrapper = shallow(<FirstApp portMessage={portMessage} portNumber={portNumber}/>)
        const paragraphText = wrapper.find('p').text();
        expect(paragraphText).toBe(`${portMessage} ${portNumber}`);
    })
});