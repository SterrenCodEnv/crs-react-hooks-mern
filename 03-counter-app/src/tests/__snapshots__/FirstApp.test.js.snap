// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in FirstApp.js Show data JSON 1`] = `
<Fragment>
  <h2>
    Data:
  </h2>
  <pre>
    {
    "firstName": "Juan",
    "lasName": "Sterren",
    "email": "sterrenjuanignacio@gmail.com",
    "age": 30,
    "job": "Software Developer"
}
  </pre>
  <p>
    Listening on port
     
    3000
  </p>
</Fragment>
`;
